src/Examine/LuceneEngine/Providers/LuceneIndex.cs

```diff
@@ -459,14 +467,45 @@ private void CreateNewIndex(Directory dir)
                     //unlock it!
                     IndexWriter.Unlock(dir);
                 }
+
                 //create the writer (this will overwrite old index files)
                 writer = new IndexWriter(dir, FieldAnalyzer, true, IndexWriter.MaxFieldLength.UNLIMITED);
+                if (_directory is ExamineDirectory examineDirectory)
+                {
+                 
+                    if (examineDirectory.GetDeletionPolicy() != null)
+                    {
+                        writer.Dispose();
+                        writer = new IndexWriter(dir, FieldAnalyzer, true, examineDirectory.GetDeletionPolicy(),
```

Comment:
Why is this calling `new IndexWriter` twice? This method is just for overwriting/creating a new index. The `writer` instance here is always disposed in the finally block of this method. This seems a bit strange since it's calling 

`writer = new IndexWriter(dir, FieldAnalyzer, true, IndexWriter.MaxFieldLength.UNLIMITED);`

to overwrite the index

then you are calling `writer = new IndexWriter(dir, FieldAnalyzer, true, examineDirectory.GetDeletionPolicy(), IndexWriter.MaxFieldLength.UNLIMITED);` 

to overwrite the index again?

Reply:
@Shazwazza I get into weird issues if I didn't called that second time with deletion policy wasn't later working, not sure why, but yeah it should be only once.